src/execution_plan/ops/op_create.c

```diff
@@ -142,8 +142,12 @@ static Record CreateConsume(OpBase *opBase) {
 	 * index R/W lock, as such free all execution plan operation up the chain. */
 	if(child) OpBase_PropagateFree(child);
 
+	// Lock everything.
+	if(!QueryCtx_LockForCommit()) return NULL;
 	// Create entities.
 	CommitNewEntities(&op->pending);
+	// Release lock.
+	QueryCtx_UnlockCommit((OpBase *)op);
```

Comment:
This logic can be moved to the shared `CommitNewEntities` function and removed from here and MergeCreate, but I have no strong preference if you think this is cleaner.

Reply:
changed